### PR TITLE
Fix flaky tests in `navigation.spec.js` and other tests related to the Post Editor Template mode

### PR DESCRIPTION
**Pull request by DAreRodz**
## What?
Fixes flaky tests in the `navigation.spec.js` file belonging to the 'Navigation block - Frontend interactivity' test group.

- https://github.com/WordPress/gutenberg/issues/51537
- https://github.com/WordPress/gutenberg/issues/51543
- https://github.com/WordPress/gutenberg/issues/51348

Other tests were failing, this time related to the `Post Editor Template mode` (see [post-editor-template-mode.spec.js](https://github.com/WordPress/gutenberg/blob/9ee7e63734892e20c76649e763bf25da525b01a6/test/e2e/specs/editor/various/post-editor-template-mode.spec.js)). I fixed them as well.

## Why?

The feature works fine, but tests are randomly failing because Playwright seems to fail with consecutive `Tab` keystrokes without a delay in between.

## How?

Replaced consecutive `page.keyboard.press` calls with `pageUtils.pressKeys`, passing a delay of 50 milliseconds.

## Testing Instructions

Automatic e2e tests should pass now.

### CODE REVIEW

**Review by cbravobernal — APPROVED** [review]
LGTM

**Comment by DAreRodz** [comment]
Ugh! There are still e2e tests that fail; they look unrelated, though...

If they do not pass after re-running them, I'll probably merge this PR anyway. 🤷 

---

EDIT: I link the failing tests for visibility
- https://github.com/WordPress/gutenberg/issues/44553

**Comment by cbravobernal** [comment]
YOLO

**Comment by DAreRodz** [comment]
I fixed other tests that were failing (also locally) related to the Post Editor Template mode.

```
Running 2 tests using 1 worker

  ✘  1 …ditor Template mode › Allow to switch to template mode, edit the template and check the result (13.1s)
  ✘  2 ….js:73:2 › Post Editor Template mode › Allow creating custom block templates in classic themes (13.1s)
```

I did so by waiting for the Editor canvas to be loaded before using the keyboard.

**Comment by noisysocks** [comment]
I rebased this so that https://github.com/WordPress/gutenberg/pull/51826 is included. Hopefully there's a ✅

**Comment by luisherranz** [comment]
@darerodz the Puppeteer test finally passed, so even if that one is still flaky, I'm going to merge this to unblock `trunk`.

Great work finding out the problem with these ones! 🎉

**Comment by luisherranz** [comment]
Oh, I missed the previous @noisysocks' [comment](https://github.com/WordPress/gutenberg/pull/51790#issuecomment-1603646719). Thanks for the other fix, Robert 🙂

File: test/e2e/specs/editor/blocks/navigation.spec.js
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/navigation.spec.js:996]: I love that it solves the issue and I have no problem merging this. I'm just wondering if there's an actual bug that makes us use `delay`, which is a last resort? It feels weird to me that Playwright has no problem tabbing through elements in the editor but is "too fast" in the frontend 🤔 . Maybe there's something obvious I'm missing though!

**Pull request merged**
